### PR TITLE
Fix: Reduce visibility

### PR DESCRIPTION
**Pull request by localheinz**
### What is the reason for this PR?

- [x] reduces the visibility of a `public` method to `private`

💁‍♂️ The method is only used internally and the containing class is marked as experimental.

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)

### Review checklist

- [x] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

File: src/Faker/Container/ContainerBuilder.php [hunk 1]
```diff
@@ -48,11 +48,7 @@ public function build(): ContainerInterface
         return new Container($this->definitions);
     }
 
-    /**
-     * Get an array with extension that represent the default English
-     * functionality.
-     */
```
Comment (localheinz) [inline src/Faker/Container/ContainerBuilder.php:54]: Given that we reduce the visibility, the DocBlock becomes obsolete.

Comment (bram-pkg) [reply src/Faker/Container/ContainerBuilder.php:54]: Isn't this method needed if people want to extend the container and also have the default extensions merged in?

Comment (localheinz) [reply src/Faker/Container/ContainerBuilder.php:54]: Hmm, I am not sure - would be great to know if anyone even uses this method.

Here's what I am aiming for:

```diff
diff --git a/src/Faker/Container/Container.php b/src/Faker/Container/Container.php
index 9b361845..9460b17a 100644
--- a/src/Faker/Container/Container.php
+++ b/src/Faker/Container/Container.php
@@ -87,7 +87,7 @@ final class Container implements ContainerInterface
     {
         if (is_callable($definition)) {
             try {
-                return $definition();
+                return $definition($this);
             } catch (\Throwable $e) {
                 throw new ContainerException(
                     sprintf('Error while invoking callable for "%s"', $id),
diff --git a/src/Faker/Container/ContainerBuilder.php b/src/Faker/Container/ContainerBuilder.php
index 029d6f20..cc52edcb 100644
--- a/src/Faker/Container/ContainerBuilder.php
+++ b/src/Faker/Container/ContainerBuilder.php
@@ -51,14 +51,22 @@ final class ContainerBuilder
     private static function defaultExtensions(): array
     {
         return [
-            BarcodeExtension::class => Core\Barcode::class,
+            BarcodeExtension::class => static function (ContainerInterface $container): BarcodeExtension {
+                return new Core\Barcode($container->get(NumberExtension::class));
+            },
             BloodExtension::class => Core\Blood::class,
-            ColorExtension::class => Core\Color::class,
+            ColorExtension::class => static function (ContainerInterface $container): ColorExtension {
+                return new Core\Color($container->get(NumberExtension::class));
+            },
             DateTimeExtension::class => Core\DateTime::class,
             FileExtension::class => Core\File::class,
             NumberExtension::class => Core\Number::class,
-            VersionExtension::class => Core\Version::class,
-            UuidExtension::class => Core\Uuid::class,
+            VersionExtension::class => static function (ContainerInterface $container): VersionExtension {
+                return new Core\Version($container->get(NumberExtension::class));
+            },
+            UuidExtension::class => static function (ContainerInterface $container): UuidExtension {
+                return new Core\Uuid($container->get(NumberExtension::class));
+            },
         ];
     }
 
diff --git a/src/Faker/Core/Barcode.php b/src/Faker/Core/Barcode.php
index a85420be..4dc0d7d1 100644
--- a/src/Faker/Core/Barcode.php
+++ b/src/Faker/Core/Barcode.php
@@ -14,9 +14,9 @@ final class Barcode implements Extension\BarcodeExtension
 {
     private Extension\NumberExtension $numberExtension;
 
-    public function __construct(Extension\NumberExtension $numberExtension = null)
+    public function __construct(Extension\NumberExtension $numberExtension)
     {
-        $this->numberExtension = $numberExtension ?: new Number();
+        $this->numberExtension = $numberExtension;
     }
 
     private function ean(int $length = 13): string
diff --git a/src/Faker/Core/Color.php b/src/Faker/Core/Color.php
index bd948190..87a1b33a 100644
--- a/src/Faker/Core/Color.php
+++ b/src/Faker/Core/Color.php
@@ -54,9 +54,9 @@ final class Color implements Extension\ColorExtension
         'Turquoise', 'Violet', 'Wheat', 'White', 'WhiteSmoke', 'Yellow', 'YellowGreen',
     ];
 
-    public function __construct(Extension\NumberExtension $numberExtension = null)
+    public function __construct(Extension\NumberExtension $numberExtension)
     {
-        $this->numberExtension = $numberExtension ?: new Number();
+        $this->numberExtension = $numberExtension;
     }
 
     /**
diff --git a/src/Faker/Core/Uuid.php b/src/Faker/Core/Uuid.php
index d1db1b22..47ebb4a4 100644
--- a/src/Faker/Core/Uuid.php
+++ b/src/Faker/Core/Uuid.php
@@ -11,10 +11,10 @@ final class Uuid implements Extension\UuidExtension
 {
     private Extension\NumberExtension $numberExtension;
 
-    public function __construct(Extension\NumberExtension $numberExtension = null)
+    public function __construct(Extension\NumberExtension $numberExtension)
     {
 
-        $this->numberExtension = $numberExtension ?: new Number();
+        $this->numberExtension = $numberExtension;
     }
 
     public function uuid3(): string
diff --git a/src/Faker/Core/Version.php b/src/Faker/Core/Version.php
index 8863c480..e2de9c5e 100644
--- a/src/Faker/Core/Version.php
+++ b/src/Faker/Core/Version.php
@@ -18,10 +18,10 @@ final class Version implements Extension\VersionExtension
      */
     private array $semverCommonPreReleaseIdentifiers = ['alpha', 'beta', 'rc'];
 
-    public function __construct(Extension\NumberExtension $numberExtension = null)
+    public function __construct(Extension\NumberExtension $numberExtension)
     {
 
-        $this->numberExtension = $numberExtension ?: new  Number();
+        $this->numberExtension = $numberExtension;
     }
 
     /**
diff --git a/test/Faker/Core/ColorTest.php b/test/Faker/Core/ColorTest.php
index 13183c89..b06bf255 100644
--- a/test/Faker/Core/ColorTest.php
+++ b/test/Faker/Core/ColorTest.php
@@ -5,45 +5,46 @@ declare(strict_types=1);
 namespace Faker\Test\Core;
 
 use Faker\Core\Color;
+use Faker\Core\Number;
 use Faker\Test\TestCase;
 
 final class ColorTest extends TestCase
 {
     public function testHexColor(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         self::assertMatchesRegularExpression('/^#[a-f0-9]{6}$/i', $color->hexColor());
     }
 
     public function testSafeHexColor(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         self::assertMatchesRegularExpression('/^#[a-f0-9]{6}$/i', $color->safeHexColor());
     }
 
     public function testRgbColorAsArray(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         self::assertCount(3, $color->rgbColorAsArray());
     }
 
     public function testRgbColor(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         $regexp = '([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])';
         self::assertMatchesRegularExpression('/^' . $regexp . ',' . $regexp . ',' . $regexp . '$/i', $color->rgbColor());
     }
 
     public function testRgbCssColor(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         $regexp = '([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])';
         self::assertMatchesRegularExpression('/^rgb\(' . $regexp . ',' . $regexp . ',' . $regexp . '\)$/i', $color->rgbCssColor());
     }
 
     public function testRgbaCssColor(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         $regexp = '([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])';
         $regexpAlpha = '([01]?(\.\d+)?)';
         self::assertMatchesRegularExpression('/^rgba\(' . $regexp . ',' . $regexp . ',' . $regexp . ',' . $regexpAlpha . '\)$/i', $color->rgbaCssColor());
@@ -51,19 +52,19 @@ final class ColorTest extends TestCase
 
     public function testSafeColorName(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         self::assertMatchesRegularExpression('/^[\w]+$/', $color->safeColorName());
     }
 
     public function testColorName(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         self::assertMatchesRegularExpression('/^[\w]+$/', $color->colorName());
     }
 
     public function testHslColor(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         $regexp360 = '(?:36[0]|3[0-5][0-9]|[12][0-9][0-9]|[1-9]?[0-9])';
         $regexp100 = '(?:100|[1-9]?[0-9])';
         self::assertMatchesRegularExpression('/^' . $regexp360 . ',' . $regexp100 . ',' . $regexp100 . '$/', $color->hslColor());
@@ -71,7 +72,7 @@ final class ColorTest extends TestCase
 
     public function testHslColorArray(): void
     {
-        $color = new Color();
+        $color = new Color(new Number());
         self::assertCount(3, $color->hslColorAsArray());
     }
 }
diff --git a/test/Faker/Core/UuidTest.php b/test/Faker/Core/UuidTest.php
index a0bedce0..03cae0b7 100644
--- a/test/Faker/Core/UuidTest.php
+++ b/test/Faker/Core/UuidTest.php
@@ -2,6 +2,7 @@
 
 namespace Faker\Test\Core;
 
+use Faker\Core\Number;
 use Faker\Core\Uuid;
 use Faker\Test\TestCase;
 
@@ -9,14 +10,14 @@ final class UuidTest extends TestCase
 {
     public function testUuidReturnsUuid(): void
     {
-        $instance = new Uuid();
+        $instance = new Uuid(new Number());
         $uuid = $instance->uuid3();
         self::assertTrue($this->isUuid($uuid));
     }
 
     public function testUuidExpectedSeed(): void
     {
-        $instance = new Uuid();
+        $instance = new Uuid(new Number());
 
         if (pack('L', 0x6162797A) == pack('N', 0x6162797A)) {
             self::markTestSkipped('Big Endian');
```

**Review by bram-pkg — APPROVED** [review]

**Comment by localheinz** [comment]
Thank you, @bram-pkg!

**Pull request merged**